integration/tests/basic_functionality/negative/neg_deploy_with_invalid_r10k_yaml.rb

```diff
@@ -0,0 +1,46 @@
+confine :except, :platform => 'solaris-10'
+require 'git_utils'
+require 'r10k_utils'
+require 'master_manipulator'
+test_name 'CODEMGMT-84 - C59271 Attempt to Deploy with Invalid r10k Config'
+
+#Init
+env_path = on(master, puppet('config print environmentpath')).stdout.rstrip
+git_repo_path = '/git_repos'
+git_control_remote = File.join(git_repo_path, 'environments.git')
+
+r10k_config_path = '/etc/r10k.yaml'
+r10k_config_bak_path = "#{r10k_config_path}.bak"
+
+#In-line files
+r10k_conf = <<-CONF
+sources:
+  broken:
+    basedir: "#{env_path}"
+    remote: "#{git_control_remote}"
+CONF
+
+#Verification
+error_message_regex = /ERROR.*can\'t\ convert\ nil\ into\ String/
+
+#Teardown
+teardown do
+  step 'Restore Original "r10k" Config'
+  on(master, "mv #{r10k_config_bak_path} #{r10k_config_path}")
+end
+
+#Setup
+step 'Backup Current "r10k" Config'
+on(master, "mv #{r10k_config_path} #{r10k_config_bak_path}")
+
+step 'Update the "r10k" Config'
+create_remote_file(master, r10k_config_path, r10k_conf)
+
+step 'Invalidate the "r10k" Config'
+on(master, "sed -i 's/basedir/dir/' #{r10k_config_path}") 
```

Comment:
This really isn't necessary since you can corrupt the r10k config in the declaration of the "r10k_conf" variable.